datadog_checks_base/datadog_checks/base/utils/db/schemas.py

```diff
@@ -0,0 +1,197 @@
+# (C) Datadog, Inc. 2025-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, TypedDict
+
+from datadog_checks.base.utils.serialization import json
+
+from .utils import now_ms
+
+if TYPE_CHECKING:
+    from datadog_checks.base.checks.db import DatabaseCheck
+
+try:
+    import datadog_agent  # type: ignore
+except ImportError:
+    from datadog_checks.base.stubs import datadog_agent
+
+
+class DatabaseInfo(TypedDict):
+    name: str
+
+
+# The schema collector sends lists of DatabaseObjects to the agent
+# DBMS subclasses may add additional fields to the dictionary
+class DatabaseObject(TypedDict):
+    name: str
+
+
+# Common configuration for schema collector
+# Individual DBMS implementations should map their specific
+# configuration to this type
+class SchemaCollectorConfig:
+    def __init__(self):
+        self.collection_interval = 3600
+        self.payload_chunk_size = 10_000
+
+
+class SchemaCollector(ABC):
+    """
+    Abstract base class for DBM schema collectors.
+
+    Attributes:
+        _collection_started_at (int): Timestamp in whole milliseconds
+            when the current collection started.
+    """
+
+    _collection_started_at: int | None = None
+
+    def __init__(self, check: DatabaseCheck, config: SchemaCollectorConfig):
+        self._check = check
+        self._log = check.log
+        self._config = config
+        self._dbms = check.__class__.__name__.lower()
+        if self._dbms == 'postgresql':
+            # Backwards compatibility for metrics namespacing
+            self._dbms = 'postgres'
```

Comment:
I think we can move this onto `DatabaseCheck` itself or have `DatabaseCheck.dbms` abstract method. I can see this being useful elsewhere